primitives/consensus/common/src/import_queue/basic_queue.rs

```diff
@@ -405,22 +361,51 @@ fn import_many_blocks<B: BlockT, V: Verifier<B>, Transaction>(
 		}
 
 		if import_result.is_ok() {
-			trace!(target: "sync", "Block imported successfully {:?} ({})", block_number, block_hash);
+			trace!(
+				target: "sync",
+				"Block imported successfully {:?} ({})",
+				block_number,
+				block_hash,
+			);
 			imported += 1;
 		} else {
 			has_error = true;
 		}
 
 		results.push((import_result, block_hash));
 
-		// Notifies the current task again so that we re-execute this closure again for the next
-		// block.
-		if delay_between_blocks != Duration::new(0, 0) {
-			waiting = Some(Delay::new(delay_between_blocks));
+		if delay_between_blocks != Duration::default() && !has_error {
```

Comment:
This `!has_error` is a behavior change, but it makes sense, there's no point in delaying cancellations 👍 (in practice we don't use this delay anyway).

Reply:
In the end, after the first error has occurred, we would always return `Err(Canceled)` anyway.